battle/battle_package/algorithm.py

```diff
@@ -2,104 +2,97 @@
 from .request import Request
 
 
-class BattleAlgo:
+class Battle:
 
-	@staticmethod
-	# The function takes care of defining a json containing the attacker's parameters
-	def define_attack(report):
-		attacker = {}
-		for key, value in report.items():
-			if key == "attacker":
-				attacker = value
-		return attacker
-
-	@staticmethod
-	# The function takes care of defining a json containing the parameters of the defender
-	def define_defender(report):
-		defender = {}
-		for key, value in report.items():
-			if key == "defender":
-				defender = value
-		return defender
+	def __init__(self, request):
+		self.attacker = request.challengers_armies['attacker']  # parameters of the attacker
+		self.defender = request.challengers_armies['defender']  # parameters of the defender
+		self.focus, self.advantage = Battle.deploy(self)
```

Comment:
is `Battle.deploy(self)` the same as `self.deploy()`  ?

Reply:
`self.deploy()` is the correct syntax for instance methods